In [6]:
import pandas as pd

In [10]:
student_data = {
    "Name":["Jane","James","Joseph","Josephine","andrew"],
    "Science":[89,98,79,79,69],
    "Maths":[91,86,95,69,89],
    "Gender":["female","Male","Male","Male","Female"]

}
students = pd.DataFrame(student_data)
students

,Name,Science,Maths,Gender
0,Jane,89,91,female
1,James,98,86,Male
2,Joseph,79,95,Male
3,Josephine,79,69,Male
4,andrew,69,89,Female


In [11]:
students["Maths"].mode()

0    69
1    86
2    89
3    91
4    95
Name: Maths, dtype: int64

find the mean science mark and explain your result
find the median of the science marks and list the student/students attached to that mark

In [12]:
students["Science"].mean()

np.float64(82.8)

this means that on average, the students scored 82.8

In [13]:
students["Science"].median()

np.float64(79.0)

joseph and josephine are the students who scored the science median 

exercise 2
find the spread of the science marks using variances
what is the range of the science marks and comment on your answer

In [14]:
students["Science"].var()

np.float64(122.20000000000002)

In [15]:
students["Science"].max()-students["Science"].min()

np.int64(29)

this means that the difference between the highest and the lowest science marks is 29 marks

measures of position
-percentiles
-quartiles
-deciles etc

In [26]:
## Quartiles
### lower quantile
lower_quantile = students["Maths"].quantile(0.25)
lower_quantile = lower_quantile

In [20]:
### upper quantile
upper_quantile = students["Maths"].quantile(0.75)
upper_quantile = upper_quantile

In [21]:
IQR = upper_quantile - lower_quantile
IQR

np.float64(5.0)

PERCENTILES

In [22]:
import numpy as np

In [25]:
lower_quantile2 = np.percentile(students["Maths"], 25)
lower_quantile2

np.float64(86.0)

90th percentile for maths

In [27]:
ninty_th = np.percentile(students["Maths"], 90)
ninty_th

np.float64(93.4)